### PR TITLE
style: Account for cousin sharing in the revalidation logic. 

### PR DESCRIPTION
**Pull request by emilio**
Fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1361013

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16670)

### CODE REVIEW

**Comment by emilio** [comment]
r? @bholley 

cc @heycam

File: components/style/stylist.rs [hunk 1]
```diff
@@ -375,10 +376,7 @@ impl Stylist {
     #[inline]
     fn note_for_revalidation(&mut self, selector: &Selector<SelectorImpl>) {
         if needs_revalidation(selector) {
-            // For revalidation, we can skip everything left of the first ancestor
-            // combinator.
-            let revalidation_sel = selector.inner.slice_to_first_ancestor_combinator();
```
Comment (bholley) [inline components/style/stylist.rs]: We should probably get rid of the slice_to_first_ancestor_combinator() API while we're at it. I don't think we use it for anything else.

**Review by bholley — COMMENTED** [review]
r=me with that changed. Thanks for fixing this!

File: components/style/stylist.rs (hunk 1 above)
Comment (emilio) [inline components/style/stylist.rs]: Oh, I'd bet I had done that... Probably stashed it on the way, huh. Will do

**Pull request merged**